js/middlewares/analytics.js

```diff
@@ -90,10 +93,12 @@ const actionTracking = () => next => action => {
 }
 
 // gets the current screen from navigation state
-function getCurrentRouteName(navigationState) {
+// TODO: Need to be fixed
+function getCurrentRouteName(navigationState): ?string {
```

Comment:
this may be a little better:

```js
import { type NavigationState, type NavigationLeafRoute } from 'react-navigation'

// gets the current screen from navigation state
function getCurrentRouteName(navNode: NavigationState | NavigationLeafRoute): ?string {
  if (!navNode) {
    return null
  }

  if(typeof navNode.routeName === 'string') {
    // navNode is a NavigationLeafRoute
    return navNode.routeName;
  }

  // navNode is a NavigationState
  const navState = ((navNode: any): NavigationState)
  const route = navState.routes[navState.index]
  return getCurrentRouteName(route)

}
```

Flow should be able to figure out that `navNode` is a `NavigationState`, but it doesn't... TypeScript is much better...